File: 2488-divide-intervals-into-minimum-number-of-groups/divide-intervals-into-minimum-number-of-groups.cpp
class Solution {
public:
    int minGroups(vector<vector<int>>& intervals) {
        vector<int> start;
        vector<int> end;
        for (vector<int> temp: intervals){
            start.push_back(temp[0]);
            end.push_back(temp[1]);
        }
        sort(start.begin(), start.end());
        sort(end.begin(), end.end());
        int count = 0, result = 0, i = 0, j = 0;
        while (i < intervals.size() && j < intervals.size()){
            if (start[i] <= end[j]){
                count += 1;
                i += 1;
            }
            else{
                count -= 1;
                j += 1;
            }
            result = max(result, count);
        }
        return result;
    }
};

File: 2488-divide-intervals-into-minimum-number-of-groups/divide-intervals-into-minimum-number-of-groups.java
class Solution {
    public int minGroups(int[][] intervals) {
        int[] start = new int[intervals.length];
        int[] end = new int[intervals.length];
        for (int i = 0; i < intervals.length; i++){
            start[i] = intervals[i][0];
            end[i] = intervals[i][1];
        }
        Arrays.sort(start);
        Arrays.sort(end);
        int count = 0, result = 0, i = 0, j = 0;
        while (i < intervals.length && j < intervals.length){
            if (start[i] <= end[j]){
                count += 1;
                i += 1;
            }
            else{
                count -= 1;
                j += 1;
            }
            result = Math.max(result, count);
        }
        return result;
    }
}